### ISSUE TITLE
Equal sign in img tag converts to dash

### ISSUE THREAD

**Issue by asmith20002** [SimpleMachines/SMF#7528]
Using fresh copy of SMF 2.1.2.

If an img tag has an equal sign in it, the first apperance of it, converts into dash.

Example:
Try posting this string into your forums:
`[img]http://localhost/index.php?action=test;file=image.jpg[/img]`

the part `action=test` turns into `action-test`

**Comment by m4z** [comment]
~~Equal signs cannot appear in URLs (this way, compare the `action=test` part); according to https://en.wikipedia.org/wiki/Percent-encoding#Reserved_characters they should be converted to `%3D` but apparently SMF uses something more readable.~~
Sigh. It's late, sorry for the noise.

**Comment by jdarwood007** [comment]
I seem to recall this in 1.1 or 2.0...

**Comment by sbulen** [comment]
This only happens on an "action=" found in your image URL.  All other uses of = test out fine & are not affected.
https://github.com/SimpleMachines/SMF/blob/6d8394d172041ca1bec3ba249f922462ace8bc78/Sources/Subs.php#L2715

Which...  Shouldn't be there???

Do you have a use case that has a valid _image_ URL with "action=" in it?  (E.g., an image sharing site that requires 'action=' for a parameter?)

**Comment by asmith20002** [comment]
Actually I do. I have written a custom SMF mod which works on images and it gets injected to the main action array via integration hook.

The "action" parameter is not uncommon in URLs too, but let's talk about SMF alone. Shouldn't it check for `action=` which are located in the same boardurl? and even then, lots of SMF forums use URL rewrites. so they may not even include `action=` and simply bypass that logic. I mean that section of code is not much of security to begin with anyway.

**Comment by sbulen** [comment]
Why not use a different parameter in your mod, especially since action= has a very specific meaning in SMF?

**Comment by asmith20002** [comment]
My production site changes `example.com/index.php?action=image;sa=process` into `example.com/image/process`
So that's not a deal breaker for me.

SMF aside, the action parameter could be in some image hosting service as you mentioned. All in all I don't think that section of code is an effective solution. When we put another non-image SMF URL inside an img tag, how is that dangerous? because the same can be achieved with any URL rewrite which doesn't include the action parameter. There are mod rewrites on SMF which changes `action=something` into `action.something` or `sa=something` into `sa.something` so that section of security check can easily be bypassed.

**Comment by sbulen** [comment]
Yes, it could be made tighter, e.g., check for the domain name/path from the $boardurl and only apply the edit then.
